src/vivarium_gates_nutrition_optimization/components/pregnancy.py

```diff
@@ -145,27 +142,38 @@ def Pregnancy():
     )
 
 
-def get_partial_term_probability_data(builder: Builder) -> pd.DataFrame:
+def get_birth_outcome_probabilities(builder: Builder) -> pd.DataFrame:
     asfr = builder.data.load(data_keys.PREGNANCY.ASFR).set_index(ARTIFACT_INDEX_COLUMNS)
     sbr = (
         builder.data.load(data_keys.PREGNANCY.SBR)
         .set_index("year_start")
         .drop(columns=["year_end"])
         .reindex(asfr.index, level="year_start")
     )
+
     raw_incidence_miscarriage = builder.data.load(
         data_keys.PREGNANCY.RAW_INCIDENCE_RATE_MISCARRIAGE
     ).set_index(ARTIFACT_INDEX_COLUMNS)
     raw_incidence_ectopic = builder.data.load(
         data_keys.PREGNANCY.RAW_INCIDENCE_RATE_ECTOPIC
     ).set_index(ARTIFACT_INDEX_COLUMNS)
 
-    return (
-        (raw_incidence_miscarriage + raw_incidence_ectopic)
-        / (
-            asfr
-            + asfr.multiply(sbr["value"], axis=0)
-            + raw_incidence_miscarriage
-            + raw_incidence_ectopic
-        )
+    total_incidence = (
+        asfr
+        + asfr.multiply(sbr["value"], axis=0)
+        + raw_incidence_ectopic
+        + raw_incidence_miscarriage
+    )
+
+    partial_term = (raw_incidence_ectopic + raw_incidence_miscarriage) / total_incidence
+    partial_term["pregnancy_outcome"] = models.PARTIAL_TERM_OUTCOME
+    live_births = asfr / total_incidence
+    live_births["pregnancy_outcome"] = models.LIVE_BIRTH_OUTCOME
+    stillbirths = asfr.multiply(sbr["value"], axis=0) / total_incidence
+    stillbirths["pregnancy_outcome"] = models.STILLBIRTH_OUTCOME
+    probabilities = pd.concat([partial_term, live_births, stillbirths]).fillna(0)
```

Comment:
The `fillna(0)` concerns me. As discussed, try removing it and see if it works.

Reply:
it seems to work fine without it